src/Stan/Report/Html.hs

```diff
@@ -38,12 +49,55 @@ stanMain an = main_
       # divIdClassH "Analysis Info" "" ()
       )
     # divIdClassH "Graphs" "" (p_ "Maybe later")
-    # divIdClassH "Observations" "" (pre_ an)
+    # divIdClassH "Observations" "" (stanObservations an)
     # divIdClassH "Configurations" "" (p_ "Later")
     # divIdClassH "Summary" "" (p_ "Later")
     # divIdClassH "Inspections" "" (p_ "Later")
     )
 
+stanObservations Analysis{..} =
+    map stanPerFile
+    $ filter (not . null . fileInfoObservations)
+    $ Map.elems analysisFileMap
+
+stanPerFile FileInfo{..} = divIdClass "" ""
+    ( h3_ (toText fileInfoPath)
+    # p_ ("Module: " # maybe mempty (unModuleName . observationModuleName) (S.safeHead fileInfoObservations))
+    # p_ ("Lines of Code:" <> show fileInfoLoc)
+    # divClass "observations"
+      ( strong_ "Observations" # map stanObservation (toList fileInfoObservations))
+    )
+
+stanObservation o@Observation{..} = divIdClass (unId observationId) "observation"
+    ( general
+    # pre_ (unlines $ prettyObservationSource False o)
+    # if null solutions then Nothing else Just $ divClass "solutions"
```

Comment:
This sounds like a function, that can be helpful in many places 🤔 Need to think about the name, and probably we can add it to `relude` as well 🙂 I feel like something like this should exist.

```
??? :: Bool -> a -> Maybe a
```

Reply:
It could be just `memptyIfTrue` from `relude` if the scary type we have was a semigroup...